tools/generator-terraform/generator/mappings/assignment_direct_createorupdate_test.go

```diff
@@ -3496,3 +3496,289 @@ func TestDirectAssignment_CreateOrUpdate_Zones_OptionalToOptional(t *testing.T)
+		mapping := resourcemanager.FieldMappingDefinition{
+			Type: resourcemanager.DirectAssignmentMappingDefinitionType,
+			DirectAssignment: &resourcemanager.FieldMappingDirectAssignmentDefinition{
+				SchemaModelName: "FromModel",
+				SchemaFieldPath: "Labels",
+				SdkFieldPath:    "Labels",
+				SdkModelName:    "ToModel",
+			},
+		}
+		schemaModel := resourcemanager.TerraformSchemaModelDefinition{
+			Fields: map[string]resourcemanager.TerraformSchemaFieldDefinition{
+				"Labels": {
+					ObjectDefinition: resourcemanager.TerraformSchemaFieldObjectDefinition{
+						Type: resourcemanager.TerraformSchemaFieldTypeDictionary,
+						NestedObject: &resourcemanager.TerraformSchemaFieldObjectDefinition{
+							Type: v.schemaModelFieldType,
+						},
+					},
+					HclName:  "labels",
+					Required: true,
+				},
+			},
+		}
+		sdkModel := resourcemanager.ModelDetails{
+			Fields: map[string]resourcemanager.FieldDetails{
+				"Labels": {
+					JsonName: "labels",
+					ObjectDefinition: resourcemanager.ApiObjectDefinition{
+						Type: resourcemanager.DictionaryApiObjectDefinitionType,
+						NestedItem: &resourcemanager.ApiObjectDefinition{
+							Type: v.sdkFieldType,
+						},
+					},
+					Required: true,
+				},
+			},
+		}
+		actual, err := directAssignmentLine{}.assignmentForCreateUpdateMapping(mapping, schemaModel, sdkModel, nil, "sdkresource")
+		if err != nil {
+			t.Fatalf("retrieving create/update assignment mapping: %+v", err)
+		}
+		if actual == nil {
+			t.Fatalf("retrieving create/update assignment mapping: `actual` was nil")
+		}
+		testhelpers.AssertTemplatedCodeMatches(t, v.expected, *actual)
+	}
+}
+
+func TestDirectAssignment_CreateOrUpdate_Dictionary_SimpleType_RequiredToOptional(t *testing.T) {
+	// mapping a Schema Model Field (Required) to an SDK Field (Optional) for a dictionary where the value is a simple type
+
+	testData := []struct {
+		schemaModelFieldType resourcemanager.TerraformSchemaFieldType
+		sdkFieldType         resourcemanager.ApiObjectDefinitionType
+		expected             string
+	}{
+		{
+			schemaModelFieldType: resourcemanager.TerraformSchemaFieldTypeString,
+			sdkFieldType:         resourcemanager.StringApiObjectDefinitionType,
+			expected:             "output.Labels = &input.Labels",
+		},
+		{
+			schemaModelFieldType: resourcemanager.TerraformSchemaFieldTypeInteger,
+			sdkFieldType:         resourcemanager.IntegerApiObjectDefinitionType,
+			expected:             "output.Labels = &input.Labels",
+		},
+		{
+			schemaModelFieldType: resourcemanager.TerraformSchemaFieldTypeBoolean,
+			sdkFieldType:         resourcemanager.BooleanApiObjectDefinitionType,
+			expected:             "output.Labels = &input.Labels",
+		},
+		{
+			schemaModelFieldType: resourcemanager.TerraformSchemaFieldTypeFloat,
+			sdkFieldType:         resourcemanager.FloatApiObjectDefinitionType,
+			expected:             "output.Labels = &input.Labels",
```

Comment:
can we use `pointer.To` here:

```suggestion
			expected:             "output.Labels = pointer.To(input.Labels)",
		},
		{
			schemaModelFieldType: resourcemanager.TerraformSchemaFieldTypeInteger,
			sdkFieldType:         resourcemanager.IntegerApiObjectDefinitionType,
			expected:             "output.Labels = pointer.To(input.Labels)",
		},
		{
			schemaModelFieldType: resourcemanager.TerraformSchemaFieldTypeBoolean,
			sdkFieldType:         resourcemanager.BooleanApiObjectDefinitionType,
			expected:             "output.Labels = pointer.To(input.Labels)",
		},
		{
			schemaModelFieldType: resourcemanager.TerraformSchemaFieldTypeFloat,
			sdkFieldType:         resourcemanager.FloatApiObjectDefinitionType,
			expected:             "output.Labels = pointer.To(input.Labels)",
```